hack/charts/cluster-api-operator/templates/infra.yaml

```diff
@@ -1,3 +1,13 @@
+{{- define "recursivePrinter" }}
+{{- range $key, $value := . }}
+{{- if kindIs "map" $value }}
+  {{ $key }}:
+  {{- include "recursivePrinter" $value | indent 2 }}
+{{- else }}
+  {{ $key }}: {{ $value }}
+{{- end }}
```

Comment:
This function allows the chart to print out `additionalDeployments` correctly in yaml format as both deployments and manager has nested fields.